apache/netbeans
# Add Jakarta EE/Java EE support for Tomcat and TomEE

- Add identifiers for Tomcat 10 and 10.1
- Add support for TomEE versions from 1.5 to 9
- Improve support for TomEE types and correctly identify them. (Microprofile, Webprofile, etc.)
- Improve Tomcat/TomEE version enum
- Improve Tomcat/TomEE jvm platform support
- Fix the default jvm platform that is pre-selected for Tomcat/TomEE
- Improve logic for supported profiles and jvm platforms for Tomcat and TomEE
- Improve logic for getJavadoc() method for Tomcat and TomEE
- Add missing tomcat versions to some methods
- Add isAtLeast() method to TomcatVersion
- Use enum isAtLeast() method to improve logic
- Improve support for Jakarta EE JaxRS
- Add Tests for new TomEEtype logic
- Add Tests for new Tomcat version comparison
- Add Tests for new TomEE version comparison
- Use generics
- Bump javac.source to 1.8
- Add missing attributes to a dtd file

Testing:
- Full build done
- Verify successful execution of libraries and licenses Ant test
- Verify successful execution of Sigtests
- Started NetBeans and ensure the log didn't have any ERROR or new WARNINGS
- Download every version of Tomcat and TomEE and successfully:
  - Register it
  - Check correct Java SE support from Java 6 to Java 20
  - Check correct Java/Jakarta EE support from JEE 5 to JakartaEE 9.1
  - Create Web App Project
  - Deploy and run a servlet
